examples/compose-object-test-example.mjs

```diff
@@ -0,0 +1,126 @@
+const sampleRunComposeObject = async () => {
+  const tmpDir = os.tmpdir()
+
+  const bucketName = 'source-bucket'
+  // generate 100 MB buffer and write to a file.
+  const local100mbFileToBeSplitAndComposed = Buffer.alloc(100 * oneMB, 0)
+
+  const composedObjName = '_100-mb-file-to-test-compose'
+  const tmpSubDir = `${tmpDir}/compose`
+  const fileToSplit = `${tmpSubDir}/${composedObjName}`
+  const partObjNameList = []
+
+  fs.mkdir(tmpSubDir, { recursive: true }, function (err) {
+    if (err) {
+      console.log(err)
+    } else {
+      console.log('New Temp directory successfully created.')
+    }
+  })
+
+  try {
+    fs.writeFileSync(fileToSplit, local100mbFileToBeSplitAndComposed)
+    console.log('Written 100 MB File ')
+    // 100 MB split into 26 MB part size. ( just to test unequal parts ). But change as required.
+
+    const names = await splitFile.splitFileBySize(fileToSplit, 26 * oneMB)
+
+    console.log('Split and write 100 MB File(s) ', names)
+    const putPartRequests = names.map((partFileName) => {
+      const partObjName = partFileName.slice((tmpSubDir + '/').length)
+      partObjNameList.push(partObjName)
+      return s3Client.fPutObject(bucketName, partObjName, partFileName, {})
+    })
+    await Promise.all(putPartRequests)
+
+    console.log('Uploaded part Files: ', names)
+    const sourcePartObjList = partObjNameList.map((partObjName) => {
+      return new Minio.CopySourceOptions({
+        Bucket: bucketName,
+        Object: partObjName,
+      })
+    })
+
+    const destObjConfig = new Minio.CopyDestinationOptions({
+      Bucket: bucketName,
+      Object: composedObjName,
+    })
+
+    try {
+      const result = await s3Client.composeObject(destObjConfig, sourcePartObjList)
+      console.log(result)
+      console.log('Composed to a single file: ', composedObjName)
+    } catch (err) {
+      console.log('Composed to a single file: ', composedObjName)
+
+      /** Begin Clean up ***/
+      // To verify that the parts are uploaded properly, comment the below code blocks and verify
+      const sourcePartObjList = partObjNameList.map((partObjName) => {
+        return s3Client.removeObject(bucketName, partObjName)
+      })
+
+      Promise.all(sourcePartObjList)
+        .then(() => {
+          console.log('Removed source parts: ')
+
+          // Uncomment to remove the composed object itself. commented for verification.
+          /*
+    s3Client.removeObject(bucketName, composedObjName).then(()=>{
+      console.log("Clean up: Removed the composed Object ")
+    }).catch(()=>{
+      console.log("Error removing composed object", er)
+    })
+    */
+        })
+        .catch((er) => {
+          console.log('Error removing parts used in composing', er)
+        })
+
+      /** End Clean up **/
+
+      // Clean up generated parts locally
+      fs.rmSync(tmpSubDir, { recursive: true, force: true })
+      console.log('Clean up temp parts directory : ')
```

Comment:
Looks like all of this is under the `catch` block, causing the individual parts not to be cleaned up after the `composeObject` operation is done. Is this expected? If not, I think we can move this out of the `catch` block

Reply:
I have updated. thank you. :+1: